Tensile/TensileLibLogicToYaml.py

```diff
@@ -0,0 +1,319 @@
+         problemTypeYamlData.append("      {}: {}\n".format(problemTypeKey, problemTypeValue))
+         continue
+      if problemTypeKey in defaultProblemType:
+         if problemTypeValue != defaultProblemType[problemTypeKey]:
+            problemTypeYamlData.append("      {}: {}\n".format(problemTypeKey, problemTypeValue))
+            
+    return ''.join(str(x) for x in problemTypeYamlData)
+            
+def formForkParametersDataForNonMI(currentIndexSolution):
+    forkParametersYamlData = []
+    
+    forkParametersYamlData.append("    - # BenchmarkProblemSizeGroup\n") 
+    forkParametersYamlData.append("      InitialSolutionParameters:\n") 
+    forkParametersYamlData.append("      BenchmarkCommonParameters:\n") 
+    
+    forkParametersYamlData.append("      ForkParameters:\n")
+    macInstruction = ""
+    for forkKey, forkValue in currentIndexSolution.items():
+      if forkKey == "MACInstruction":
+        if forkValue == "MAC": #This is a bug in the library logic file
+          macInstruction ="         - {}: ['MAD']\n".format(forkKey)
+          forkParametersYamlData.append(macInstruction)
+          continue
+        else: 
+          macInstruction ="         - {}: {}\n".format(forkKey,[forkValue])
+          forkParametersYamlData.append(macInstruction)
+          continue
+        
+      if forkKey != "ProblemType" and forkKey != "MatrixInstruction":
+        # Find the matching index for fork key name from list of dictionaries => defaultBenchmarkCommonParameters
+        index =  next((i for i,d in enumerate(defaultBenchmarkCommonParameters) if forkKey in d), None)
+        if index != None:
+          #Strip the leading and trailing [] brackets, trim the leading and trailing spaces
+          DefaultValue = str(defaultBenchmarkCommonParameters[index][forkKey])[1:-1].strip()
+          
+          # remove the "" from the string
+          DefaultValue = re.sub("[\"\']", "", DefaultValue)
+          if str(forkValue) != DefaultValue:
+            if forkKey == "ThreadTile" or forkKey == "WorkGroup":
+              forkParametersYamlData.append("         - {}: {}\n".format(forkKey, forkValue))
+            else:
+              forkParametersYamlData.append("         - {}: {}\n".format(forkKey, [forkValue]))
+              
+    return ''.join(str(x) for x in forkParametersYamlData)
+
+def formForkParametersYamlData(currentIndexSolution, MIInstruction9Bits):
+    forkParametersYamlData = []
+    forkParametersYamlData.append("    - # BenchmarkProblemSizeGroup\n") 
+    forkParametersYamlData.append("      InitialSolutionParameters:\n") 
+    forkParametersYamlData.append("      BenchmarkCommonParameters:\n") 
+    forkParametersYamlData.append("      ForkParameters:\n")
+    
+    macInstruction = ""
+    for forkKey, forkValue in currentIndexSolution.items():
+      if forkKey == "MACInstruction":
+        if forkValue == "MAC": #This is a bug in the library logic file

```

Comment:
We do not need to append maxInstruction, right?
It looks like it is removed later.

Reply:
Thank you. if we are using MI instruction we don't need MAC instruction earlier. I removed that logic. earlier we thought both MI instruction and non MI instruction could come here.